TTS/tts/models/base_tts.py

```diff
@@ -234,6 +235,26 @@ def format_batch(self, batch: Dict) -> Dict:
             "language_ids": language_ids,
         }
 
+    def get_sampler(self, config: Coqpit, data_items: List, sampler: bool = None):
+        weights = None
+        if getattr(config, "use_language_weighted_sampler", False):
+            alpha = getattr(config, "language_weighted_sampler_alpha", 1.0)
+            print(" > Using Language weighted sampler with alpha:", alpha)
+            weights = get_language_balancer_weights(data_items) * alpha
+
+        if getattr(config, "use_speaker_weighted_sampler", False):
+            alpha = getattr(config, "speaker_weighted_sampler_alpha", 1.0)
+            print(" > Using Speaker weighted sampler with alpha:", alpha)
+            if weights is not None:
+                weights += get_speaker_balancer_weights(data_items) * alpha
+            else:
+                weights = get_speaker_balancer_weights(data_items) * alpha
+
+        if weights is not None:
+            sampler = WeightedRandomSampler(weights, len(weights))
```

Comment:
Is this DP compatible?

Reply:
Not for now

Reply:
Now it is DDP compatible :)

Reply:
Maybe rename `DistributedSamplerWithSampler` as `DistributedSamplerWrapper` ?

Reply:
Done